### ISSUE TITLE
[ISSUE] Annoying chat problems

### ISSUE THREAD

**Issue by Greenorine** [SmartlyDressedGames/Unturned-3.x-Community#2264]
_Hello there, Nelson! I fount some annoying problems with new chat. I would be very grateful if you would fix them (like many others, I think). All of the following is my personal opinion, please treat it with respect! Thank you in advance for reading my suggestions._

**Problems:**
- First of all, it's old problem about message's size: The short length of the message often does not allow you to express the completeness of your thoughts and you have to divide the messages into several, so that the interlocutor then looks for the rest and collects everything in a pile. ~~And that's not all! Because of this, rich messages feel very, very bad! You can see it on next **picture**:~~ ([Comment](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/issues/2264#issuecomment-749532346))
![image](https://user-images.githubusercontent.com/41028671/102882931-fc02cd00-4470-11eb-8f5f-f8aee7804bbc.png)

- Rich messages looks strange with \<b\> code: Previously, the messages looked great, they were clearly visible, the eye was happy. But now they look soapy, unpleasant and annoying. You see it again on picture above.

- Now the chat flips up in jerks, losing its smoothness: This is bad, because because of similar messages, there is a small embarrassment, and in General-the chat becomes wooden and not very convenient because of this.

**Suggestions:**

- Make the size of chat messages larger, or allow the server to change this size.
- Slightly work on the thickness to make the \</b\> code look concise.
- Return the smoothness of the chat.
- Ignore rich message codes when counting the number of characters. (Apologies if this was fixed in the last update)

**Comment by Charterino** [comment]
TLDR: The new UI upgrades messed up the chat by removing smooth scrolling and changing the way the font looks. Also configurable chat length limit would be great. Is there anything thats currently stopping us from removing the chat length limit?

**Comment by Trojaner** [comment]
>  Is there anything thats currently stopping us from removing the chat length limit?

Yes, Nelson still supports IMGUI which does not allow longer chat messages for some reason

**Comment by Greenorine** [comment]
> > Is there anything thats currently stopping us from removing the chat length limit?
> 
> Yes, Nelson still supports IMGUI which does not allow longer chat messages for some reason

Btw the length of the chat was previously more. Now even not very long messages are shown poorly.

**Comment by Sl4vP0weR** [comment]
Will be pretty to use custom fonts in messages

**Comment by Greenorine** [comment]
I did some tests, during which I realized that Nelson had removed the old simplified names for colors. Everything works with HEX.

**Comment by DiFFoZ** [comment]
Text Mesh Pro just doesn't support a `cyan` name color, other work fine

**Comment by Sl4vP0weR** [comment]
> Text Mesh Pro just doesn't support a `cyan` and other name colors, another works fine. Docs: http://digitalnativestudios.com/textmeshpro/docs/rich-text/#color

text replacement to hex color

**Comment by SDGNelson** [comment]
Technically sending strings up to 2048 bytes is supported after some improvements a while ago, but `serverSendMessage` only allows strings up to 255 characters. This should be enough to fill both lines of each chat entry however - the intention is that each message either from players or plugins can fill the chat entry.

I agree that the uGUI text does not look as nice/sharp as IMGUI. This is because uGUI uses distance field text rendering whereas IMGUI uses bitmaps. I did experiment a lot with tuning to make it nicer, but if anyone has suggestions for the TMP settings to make it better I am all ears.

For boldness I would encourage using it sparingly, i.e. the entire message should not be bold, only important words should be bold.

I do not think I understand the jerkyness / smoothness comments. Nothing should have changed in regards to scrolling?

**Comment by Greenorine** [comment]
> Technically sending strings up to 2048 bytes is supported after some improvements a while ago, but `serverSendMessage` only allows strings up to 255 characters. This should be enough to fill both lines of each chat entry however - the intention is that each message either from players or plugins can fill the chat entry.
> 
> I agree that the uGUI text does not look as nice/sharp as IMGUI. This is because uGUI uses distance field text rendering whereas IMGUI uses bitmaps. I did experiment a lot with tuning to make it nicer, but if anyone has suggestions for the TMP settings to make it better I am all ears.
> 
> For boldness I would encourage using it sparingly, i.e. the entire message should not be bold, only important words should be bold.
> 
> I do not think I understand the jerkyness / smoothness comments. Nothing should have changed in regards to scrolling?

Thank you for the answer. Will it be possible to overcome the chat limit in the future? 
As for scrolling - it means scrolling with the middle button.

**Comment by SDGNelson** [comment]
Trojaner is correct that the chat limit is unlikely to change in the near future because both IMGUI and uGUI need to be supported.

I just tested and scrolling with the middle mouse wheel in chat seems to work properly, but I think I see what you mean: each tick of the scroll wheel moves the window by 40 pixels, which also happens to be the height of each chat entry. This seems useful to me however.

**Comment by Greenorine** [comment]
> Trojaner is correct that the chat limit is unlikely to change in the near future because both IMGUI and uGUI need to be supported.
> 
> I just tested and scrolling with the middle mouse wheel in chat seems to work properly, but I think I see what you mean: each tick of the scroll wheel moves the window by 40 pixels, which also happens to be the height of each chat entry. This seems useful to me however.

Welp, it looks clumsy and unpleasant to the eye.

**Comment by Greenorine** [comment]
Thank you for listening to me. I hope that some of this will be changed in the future.
